### PR TITLE
Get rid of compose in integration tests and spawn containers programmatically

### PR DESCRIPTION
**Pull request by osinstom**
With this PR, integration tests dynamically spawn new Docker containers for each test case. It's useful to test more complex scenarios (e.g., crash or restarts of a datapath). 

TODO:
- [x] remove temporary Docker network
- [x] code cleaning

### CODE REVIEW

**Comment by osinstom** [comment]
retest this please

File: Makefile [hunk 1]
```diff
@@ -76,12 +76,12 @@ output:
 		.;
 	rm -rf output && mkdir output && tar -xf output.tar -C output && rm -f output.tar
 
-test-up4-integration-docker:
-	docker-compose -f test/integration/infra/docker-compose.yml rm -fsv
-	COMPOSE_DOCKER_CLI_BUILD=1 DOCKER_BUILDKIT=$(DOCKER_BUILDKIT) docker-compose -f test/integration/infra/docker-compose.yml build $(DOCKER_BUILD_ARGS)
-	docker-compose -f test/integration/infra/docker-compose.yml up -d
+test-up4-integration-docker: DOCKER_TARGETS=pfcpiface
+test-up4-integration-docker: DOCKER_TAG=integration
+test-up4-integration-docker: docker-build
```
Comment (pudelkoM) [inline Makefile:81]: what is done here? I don't think I have seen a target repeated multiple times in a Makefile before.

**Review by pudelkoM — APPROVED** [review]
lgtm

File: Makefile (hunk 1 above)
Comment (osinstom) [reply Makefile:81]: Just setting env variables and making `test-up4-integration-docker` depend on `docker-build`. See https://stackoverflow.com/a/15230658.

Comment (pudelkoM) [reply Makefile:81]: ok, that's what I expected. Thanks for the link.

**Pull request merged**